dft/src/util.rs

```diff
@@ -1,23 +1,24 @@
 use alloc::vec;
 
-use p3_field::Field;
+use p3_field::{Field, PackedValue};
 use p3_matrix::dense::RowMajorMatrix;
-use p3_matrix::Matrix;
+use p3_matrix::MatrixRowSlicesMut;
 
 /// Divide each coefficient of the given matrix by its height.
-pub(crate) fn divide_by_height<F: Field>(mat: &mut RowMajorMatrix<F>) {
+pub fn divide_by_height<F: Field>(mat: &mut impl MatrixRowSlicesMut<F>) {
     let h = mat.height();
     let h_inv = F::from_canonical_usize(h).inverse();
-    let (prefix, shorts, suffix) = unsafe { mat.values.align_to_mut::<F::Packing>() };
-    prefix.iter_mut().for_each(|x| *x *= h_inv);
-    shorts.iter_mut().for_each(|x| *x *= h_inv);
-    suffix.iter_mut().for_each(|x| *x *= h_inv);
+    for r in 0..h {
```

Comment:
Yeah I kind of made a half-hearted attempt to move `Matrix` methods into traits / use traits instead of just `RowMajorMatrix` but didn't fully commit to it in this PR, we either want `.par_rows_mut()` here or some api to mutate every single element of a matrix with the same function. I was shy to add more `Matrix` subtraits before giving it more thought